src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/SocketInput.cs

```diff
@@ -45,6 +45,19 @@ public SocketInput(MemoryPool memory, IThreadPool threadPool, IBufferSizeControl
 
         public bool IsCompleted => ReferenceEquals(_awaitableState, _awaitableIsCompleted);
 
+        public bool CheckFinOrThrow()
```

Comment:
I was thinking of making it internal. But why make it tri-state since you're either getting a bool from `CheckFinOrThrow` or an exception?